specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+| Rendering | ```RenderedAdaptiveCard render(Context, FragmentManager, AdaptiveCard, HostConfig)``` | P0 |
+| Custom Element | ```void registerRenderer(string, IBaseCardElementRenderer)``` | P1 |
+| Custom Element | ```IBaseCardElementRenderer(string)``` | P1 |
+| Custom Action | ```void registerActionRenderer(string, IBaseActionElementRenderer)``` | P1 |
+| Custom Action | ```IBaseActionElementRenderer getActionRenderer(string)``` | P1 |
+| Custom Action Layout | ```void registerActionLayoutRenderer(IActionLayoutRenderer)``` | P2 |
+| Custom Action Layout | ```IActionLayoutRenderer getActionLayoutRenderer()``` | P2 | 
+| Resource Resolvers | ```void registerOnlineImageLoader(IOnlineImageLoader)``` – deprecated | P3 |
+| Resource Resolvers | ```IOnlineImageLoader getOnlineImageLoader()``` - deprecated | P3 |
+| Resource Resolvers | ```void registerResourceResolver(string, IResourceResolver)``` | P2 |
+| Resource Resolvers | ```IResourceResolver getResourceResolver(string)``` | P2 |
+| Media Loading | ```IOnlineMediaLoader getOnlineMediaLoader()``` | P3 |
+| Media Loading | ```void registerOnlineMediaLoader(IOnlineMediaLoader)``` | P3 |
+| Feature Registration | ```void registerFeatureRegistration(FeatureRegistration)``` | P0 |
+| Feature Registration | ```FeatureRegistration getFeatureRegistration()``` | P0 |
+| Input Watcher | ```void setInputWatcher(IInputWatcher)``` | P3 |
+| Input Watcher | ```IInputWatcher getInputWatcher()``` | P3 |
+| Custom Fonts | ```void registerCustomTypeface(string, Typeface)``` | P3 |
+| Resource Information | ```RemoteResourceInformationVector GetResourceInformation()``` | P2 |
+
+### iOS
+
+iOS bindings are done with an API definition file and a Structs class where structs and enums are re-defined. In the API definition all methods and classes to be made public are included while the Structs file contains enumerations.
+
+#### ApiDefinitions
+``` C#
+// @interface ACOBaseActionElement : NSObject
+[BaseType(typeof(NSObject))]
+interface ACOBaseActionElement
+{
+    // @property ACRActionType type;
+    [Export("type", ArgumentSemantic.Assign)]
+    ACRActionType Type { get; set; }
+
+    // -(NSString *)title;
+    [Export("title")]
+    string Title { get; }
+
+    // -(NSString *)elementId;
+    [Export("elementId")]
+    string ElementId { get; }
+
+    // -(NSString *)url;
+    [Export("url")]
+    string Url { get; }
+
+    // -(NSString *)data;
+    [Export("data")]
+    string Data { get; }
+
+    // -(NSData *)additionalProperty;
+    [Export("additionalProperty")]
+    NSData AdditionalProperty { get; }
+}
+```
+
+#### Structs
+``` C#
+[Native]
+public enum ACRActionType : long
+{
+	ShowCard = 1,
+	Submit,
+	OpenUrl
+}
+```
+
+### APIs to be included in Xamarin
+
+| API Name | iOS API | Priority |
+| --- | --- | --- |
+| Parsing | ```(ACOAdaptiveCardParseResult*) fromJson: (NSString*)``` | P0 |
```

Comment:
> P0 | [](start = 71, length = 5)

Nit: Order by relative priority of implementation; the first column should be priority ideally